File: arch/src/main/java/com/github/lany192/arch/activity/SimpleActivity.java
package com.github.lany192.arch.activity;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.viewbinding.ViewBinding;

import com.github.lany192.arch.databinding.ToolbarDefaultBinding;

public abstract class SimpleActivity<VM extends ViewModel, VB extends ViewBinding>
        extends ModelActivity<VM, VB, ToolbarDefaultBinding> {

    @NonNull
    @Override
    public ToolbarDefaultBinding getToolbarBinding() {
        return ToolbarDefaultBinding.inflate(getLayoutInflater());
    }
}

File: arch/src/main/java/com/github/lany192/arch/activity/SimpleActivity.kt
package com.github.lany192.arch.activity

import androidx.lifecycle.ViewModel
import androidx.viewbinding.ViewBinding
import com.github.lany192.arch.databinding.ToolbarDefaultBinding
import com.github.lany192.binding.getBinding

abstract class SimpleActivity<VM : ViewModel, VB : ViewBinding> :
    ModelActivity<VM, VB, ToolbarDefaultBinding>() {

    override fun getToolbarBinding(): ToolbarDefaultBinding {
        return ToolbarDefaultBinding.inflate(layoutInflater)
    }

    override fun getContentBinding(): VB {
        return getClass<VB>(1).getBinding(layoutInflater)
    }
}